internal/core/window.rs

```diff
@@ -564,6 +568,74 @@ pub trait WindowHandleAccess {
 /// functions and generate a good signature.
 pub type WindowRc = Rc<Window>;
 
+/// This enum describes a low-level access to specific graphcis APIs used
+/// by the renderer.
+#[derive(Clone)]
+#[non_exhaustive]
+pub enum GraphicsAPI<'a> {
+    /// The rendering is done using OpenGL.
+    NativeOpenGL {
+        /// Use this function pointer to obtain access to the OpenGL implementation - similar to `eglGetProcAddress`.
+        get_proc_address: &'a dyn Fn(&str) -> *const std::ffi::c_void,
+    },
+    /// The rendering is done on a HTML Canvas element using WebGL.
+    WebGL {
+        /// The DOM element id of the HTML Canvas element used for rendering.
+        canvas_element_id: &'a str,
+        /// The drawing context type used on the HTML Canvas element for rendering. This is the argument to the
+        /// `getContext` function on the HTML Canvas element.
+        context_type: &'a str,
+    },
+    #[doc(hidden)]
+    Other,
```

Comment:
```suggestion
```

I don't think it is used, so we can as well leave it out, right?

Reply:
That's true, the change that adds the WebGL variant can remove this.